aten/src/ATen/native/vulkan/ops/Mm.cpp

```diff
@@ -10,44 +10,69 @@ namespace {
 using namespace api::utils;
 
 vTensor pack_weights(
-    api::Resource::Pool& pool,
-    const Tensor& weight_arg) {
-  // Pending Stephen's fix.
-  const Tensor weight = weight_arg.is_vulkan() ? weight_arg : weight_arg.vulkan();
-  return convert(weight);
+  api::Resource::Pool& pool,
+  const Tensor& weight_arg) {
+  if (weight_arg.is_vulkan()) {
+    return convert(weight_arg);
+  }
+  const Tensor weight = weight_arg.contiguous();
+  const IntArrayRef w_sizes = weight.sizes();
+  const float* const src_weight_ptr = weight.data_ptr<float>();
```

Comment:
Minor tip: Caching variables that are used one time only should not be necessary but does not hurt either.  The footnote to the previous statement is that that is the case unless you are dealing with more exotic move-aware types where binding of an l-value inhibits move construction / assignment without an explicit cast (i.e. std::move).  All is good here though.